### PR TITLE
Added authentication

### PR DESCRIPTION
**Pull request by sohan9819**
Added authentication 
- signup
- login
- private routes

Test User Credentials 
email : adarshbalika@gmail.com
password : adarshbalika

### CODE REVIEW

File: src/components/SignUpForm.jsx [hunk 1]
```diff
@@ -37,10 +80,27 @@ const SignUpForm = () => {
         </div>
       </div>
 
+      {/* <div className='input__text-group'>
```
Comment (chanchal-panpaliya) [inline src/components/SignUpForm.jsx:83]: remove unwanted code

**Review by chanchal-panpaliya — APPROVED** [review]
code looks clean with proper naming with good UI

File: src/components/SignUpForm.jsx (hunk 1 above)
Comment (sohan9819) [reply src/components/SignUpForm.jsx:83]: ok

**Review by rahul31agrawal — APPROVED** [review]
Hi 

Please correct the spelling of "Remember me" for check box.
rest looks good to merge.

**Pull request merged**